pkg/controllers/vdb/onlineupgrade_reconciler.go

```diff
@@ -290,6 +292,30 @@ func (r *OnlineUpgradeReconciler) loadUpgradeState(ctx context.Context) (ctrl.Re
 	return ctrl.Result{}, nil
 }
 
+// checkUpNodes will requeue the upgrade process if not all the main cluster nodes are up.
+func (r *OnlineUpgradeReconciler) checkUpNodes(ctx context.Context) (ctrl.Result, error) {
+	// We skip this if we have already added the new subclusters
+	if vmeta.GetOnlineUpgradeStepInx(r.VDB.Annotations) > addSubclustersInx {
+		return ctrl.Result{}, nil
+	}
+
+	mainPFacts := r.PFacts[vapi.MainCluster]
+	// All nodes in the main cluster must be up before running online upgrade
+	if mainPFacts.getUpNodeCount() != len(mainPFacts.Detail) {
```

Comment:
We want all the pods to be running before starting online upgrade. They don't all have to be up.

Reply:
It seems not working after changed to `countRunningAndInstalled()`. When the pending nodes just became running (not up yet), the upgrade process continued to duplicate the main cluster but returned VAddnode error. That running node and the new added nodes never came up.

Reply:
In that case, don't even check the number of upnode. Just call the restart reconciler, if any node is down it will start it, if no node is down it will do nothing, if any pod is not running, it will requeue.

Reply:
Thanks for the instruction. It is updated accordingly.